### PR TITLE
Use delayed_job  ~> 4.0.6

### PR DESCRIPTION
**Pull request by sunloverz**

### CODE REVIEW

**Comment by thousandsofthem** [comment]
is there any reason to lock to exact version?

**Comment by sunloverz** [comment]
it won't work with the latest version of delayed_job

**Comment by thousandsofthem** [comment]
Huh. kinda.... weird. well, how about at least `~> 3.0.5` ?

**Comment by sunloverz** [comment]
okay, I will change to `~> 3.0.5`

**Comment by sunloverz** [comment]
@thousandsofthem could you please review? Added the delay feature to run sometime in the future, now its compatible with the latest delayed_job and rails

File: lib/delayed/backend/actions.rb [hunk 1]
```diff
@@ -37,7 +48,9 @@ def find_available(worker_name, limit = 5, max_run_time = Worker.max_run_time)
             rescue Exception => e
               Delayed::Worker.logger.warn(e.message)
             end
-            return [Delayed::Backend::Ironmq::Job.new(message)] if message
+            if ready_to_run?(message)
```
Comment (thousandsofthem) [inline lib/delayed/backend/actions.rb]: i fear some small amount of delayed jobs can totally block processing, DJ would grab same message over and over again never reaching non-delayed ones. 
Why not to use native "delay" MQ thing?

Comment (sunloverz) [inline lib/delayed/backend/actions.rb]: You are right, working on it

**Comment by sunloverz** [comment]
@thousandsofthem Done!

**Comment by thousandsofthem** [comment]
@sunloverz looks good. can i merge?

**Comment by sunloverz** [comment]
@thousandsofthem yes, btw how about retrieving 5 jobs from the queue at once? https://github.com/iron-io/delayed_job_ironmq/blob/master/lib/delayed/backend/actions.rb#L34
Default limit is 5

**Comment by thousandsofthem** [comment]
> @thousandsofthem yes, btw how about retrieving 5 jobs from the queue at once? https://github.com/iron-io/delayed_job_ironmq/blob/master/lib/delayed/backend/actions.rb#L34
Default limit is 5

Would be nice to pass `n` to get()

**Comment by sunloverz** [comment]
It works slowly when passing `n` to get(), it seems it doesn't process all retrieved jobs, only one job at a time 
From delayed_job:
` We get up to 5 jobs from the db. In case we cannot get exclusive access to a job we try the next. this leads to a more even distribution of jobs across the worker processes
`

**Comment by thousandsofthem** [comment]
ok, thanks for clarification. merging

**Pull request merged**